src/app/public/modules/repeater/repeater-item.component.ts

```diff
@@ -246,62 +382,115 @@ export class SkyRepeaterItemComponent implements OnDestroy, OnInit {
+          event.preventDefault();
+          event.stopPropagation();
+        }
+        break;
+
+      case 'arrowdown':
+        if (this.keyboardReorderingEnabled) {
+          this.keyboardReorderDown();
+          event.preventDefault();
+          event.stopPropagation();
+        }
+        break;
+
+      case 'arrowleft':
+        if (this.keyboardReorderingEnabled) {
+          event.preventDefault();
+          event.stopPropagation();
+        }
+        break;
+
+      case 'arrowright':
+        if (this.keyboardReorderingEnabled) {
+          event.preventDefault();
+          event.stopPropagation();
+        }
+        break;
+
+      /* istanbul ignore next */
+      default:
+        break;
+    }
+  }
 
-      if (this.keyboardReorderingEnabled) {
-        this.reorderState = this.reorderStateDescription;
-      } else {
-        this.reorderState = this.reorderFinishText + ' ' + (this.reorderCurrentIndex + 1)  + ' ' + this.reorderInstructions;
-      }
-    } else if (key === 'escape') {
-      this.keyboardReorderingEnabled = false;
+  public onReorderHandleBlur(event: any): void {
+    this.keyboardReorderingEnabled = false;
+    this.revertReorderSteps();
+    this.reorderButtonLabel = this.reorderInstructions;
+    this.reorderState = undefined;
+  }
 
-      if (this.reorderSteps < 0) {
-        this.adapterService.moveItemDown(this.elementRef, Math.abs(this.reorderSteps));
-      } else if (this.reorderSteps > 0) {
-        this.adapterService.moveItemUp(this.elementRef, false, this.reorderSteps);
-      }
+  private slideForExpanded(animate: boolean): void {
+    this.slideDirection = this.isExpanded ? 'down' : 'up';
+  }
 
-      this.reorderButtonLabel = this.reorderCancelText + ' ' + this.reorderInstructions;
-
-      (<HTMLElement> event.target).focus();
-    } else if (this.keyboardReorderingEnabled && key.startsWith('arrow')) {
-      let direction = event.key.toLowerCase().replace('arrow', '');
-      if (direction === 'up') {
-        this.reorderCurrentIndex = this.adapterService.moveItemUp(this.elementRef);
-        this.reorderSteps--;
-        this.grabHandle.nativeElement.focus();
-        this.keyboardReorderingEnabled = true;
-        this.reorderButtonLabel = this.reorderMovedText + ' ' + (this.reorderCurrentIndex + 1);
-      } else if (direction === 'down') {
-        this.reorderCurrentIndex = this.adapterService.moveItemDown(this.elementRef);
-        this.reorderSteps++;
-        this.grabHandle.nativeElement.focus();
-        this.keyboardReorderingEnabled = true;
-        this.reorderButtonLabel = this.reorderMovedText + ' ' + (this.reorderCurrentIndex + 1);
-      }
+  private keyboardReorderUp() {
+    this.reorderCurrentIndex = this.adapterService.moveItemUp(this.elementRef);
+    this.reorderSteps--;
+    this.grabHandle.nativeElement.focus();
+    this.keyboardReorderingEnabled = true;
+    this.reorderButtonLabel = this.reorderMovedText + ' ' + (this.reorderCurrentIndex + 1);
```

Comment:
This might be cleaner if you used template strings.
```
this.reorderButtonLabel = `${this.reorderMovedText} ${this.reorderCurrentIndex + 1}`;
```

Reply:
Done.